src/nvim/syntax.c

```diff
@@ -7623,10 +7627,9 @@ int syn_name2id(const char *name)
 int syn_name2id_len(const char_u *name, size_t len)
   FUNC_ATTR_NONNULL_ALL
 {
-  char name_u[201];
+  char name_u[MAX_SYN_NAME + 1];
 
-  if (len == 0 || len > 200) {
-    // ID names over 200 chars don't deserve to be found!
+  if (len == 0 || len > MAX_SYN_NAME) {
```

Comment:
I think this condition can be removed or changed to the following `assert`. I believe this condition is no longer met by vim/vim@5e18ccc and vim/vim@f7f7aaf.

```C
assert(len != 0);
assert(len <= MAX_SYN_NAME);
```

Reply:
Actually, the API uses this path to report back the name was invalid.